crates/ignore/src/walk.rs

```diff
@@ -1587,7 +1612,12 @@ impl<'s> Worker<'s> {
                 false
             };
         if !should_skip_filesize && !should_skip_filtered {
-            self.send(Work { dent, ignore: ig.clone(), root_device });
+            self.send(Work {
+                dent,
+                ignore: ig.clone(),
+                root_device,
+                match_metadata_token,
```

Comment:
It's kind of cool that I could avoid eg lifetime issues by using the "token" approach, it can happily get popped onto a queue for getting sent across threads and then just eventually gets "resolved" at the point of "contact with the outside world"